spec/models/person_escort_record_spec.rb

```diff
@@ -4,8 +4,222 @@
+      described_class.save_with_responses!(profile_id: profile.id, version: '1.2')
+
+      expect(described_class.last.framework).to eq(framework)
+    end
+
+    it 'defaults to latest framework with latest version' do
+      framework = create(:framework, version: '1.2')
+      create(:framework, version: '1.0')
+      profile = create(:profile)
+      described_class.save_with_responses!(profile_id: profile.id)
+
+      expect(described_class.last.framework).to eq(framework)
+    end
+
+    it 'returns error if wrong version passed' do
+      create(:framework, version: '1.0')
+      profile = create(:profile)
+
+      expect { described_class.save_with_responses!(profile_id: profile.id, version: '2.2') }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+
+    it 'returns error if no framework found' do
+      profile = create(:profile)
+      expect { described_class.save_with_responses!(profile_id: profile.id) }.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it 'does not allow multiple person_escort_records on a profile' do
+      profile = create(:profile)
+      create(:framework)
+      described_class.save_with_responses!(profile_id: profile.id)
+
+      expect { described_class.save_with_responses!(profile_id: profile.id) }.to raise_error(ActiveRecord::RecordInvalid)
+    end
+
+    it 'sets initial state to in_progress' do
+      profile = create(:profile)
+      create(:framework)
+
+      expect(described_class.save_with_responses!(profile_id: profile.id).state).to eq('in_progress')
+    end
+
+    it 'creates responses for framework questions' do
+      profile = create(:profile)
+      framework = create(:framework)
+      create(:framework_question, :checkbox, framework: framework)
+      create(:framework_question, :checkbox, framework: framework)
+      person_escort_record = described_class.save_with_responses!(profile_id: profile.id)
+
+      responses = person_escort_record.framework_responses
+
+      expect(responses.count).to eq(2)
+    end
+
+    it 'sets correct response for framework questions' do
+      profile = create(:profile)
+      framework = create(:framework)
+      checkbox_question = create(:framework_question, :checkbox, framework: framework)
+      person_escort_record = described_class.save_with_responses!(profile_id: profile.id)
+
+      response = person_escort_record.framework_responses.first
+
+      expect(response).to have_attributes(
+        framework_question_id: checkbox_question.id,
+        type: 'FrameworkResponse::Array',
+      )
+    end
+
+    it 'allows access to framework question through person escort record' do
+      profile = create(:profile)
+      framework = create(:framework)
+      create(:framework_question, :checkbox, framework: framework)
+      person_escort_record = described_class.save_with_responses!(profile_id: profile.id)
+
+      questions = person_escort_record.framework_questions
+
+      expect(questions.count).to eq(1)
```

Comment:
I think it's fine to combine these two lines and is slightly easier to read

```suggestion
      expect(person_escort_record.framework_questions.count).to eq(1)
```